src/main/webapp/app/course/course-questions/course-questions.component.ts

```diff
@@ -0,0 +1,54 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { Subscription } from 'rxjs/Subscription';
+import { ActivatedRoute } from '@angular/router';
+import { StudentQuestionService } from 'app/overview/student-questions/student-question/student-question.service';
+import { StudentQuestion } from 'app/entities/student-question.model';
+import { SortService } from 'app/shared/service/sort.service';
+
+@Component({
+    selector: 'jhi-course-questions',
+    styles: ['.question-cell { max-width: 40vw; max-height: 120px; overflow: auto;}'],
+    templateUrl: './course-questions.component.html',
+})
+export class CourseQuestionsComponent implements OnInit, OnDestroy {
+    courseId: number;
+    studentQuestions: StudentQuestion[];
+
+    paramSub: Subscription;
```

Comment:
Is there a special reason for this subscription? 

I think `this.courseId = Number(this.route.snapshot.paramMap.get('courseId'));` inside of the ngOnInit would be sufficient.

Reply:
I've just seen it done with the subscription in quite a few different places, but I like your suggestion better. It's a lot cleaner. Thanks for the suggestion!